cli/azd/pkg/azsdk/deployments.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azsdk
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/azure/azure-dev/cli/azd/pkg/identity"
+)
+
+type DeploymentClientFactory func(subscriptionId string, credential azcore.TokenCredential) (DeploymentClient, error)
+
+type contextKey string
+
+const (
+	deploymentClientFactoryContextKey contextKey = "deploymentClientFactoryContextKey"
+)
+
+func GetDeploymentClientFactory(ctx context.Context) DeploymentClientFactory {
```

Comment:
Is the factory required here?  Would it have been possible to have a design where the client itself was on the context instead of a factory for it or does that make things much harder to work with?

Reply:
I was trying to avoid adding clients to the context.
There will be a different sdk client for each area that we need to create.
And I would like to avoid keeping the clients in head and referenced in the context. Mainly because we might want to create different clients, and not only one per command execution.

For example, provisioning/deploying to different subscriptions might be a thing in the future .

Anyway, I talked to Wallace and agree to follow one strategy for now and work things under azcli implementation